src/client/dfuse/pil4dfs/int_dfs.c

```diff
@@ -4818,6 +4903,35 @@ futimens(int fd, const struct timespec times[2])
 	return 0;
 }
 
+/* check the fd to be closed. If the fd pointing to something like
+ * 'socket:[xxxx]' or 'anon_inode:[eventpoll]', print out a warning
+ * since DAOS network contexts may be using such fds.
+ */
+static void
+check_fd_to_close(int fd)
+{
+	char    fd_path[64];
+	char   *path;
+	ssize_t len;
+
+	D_ALLOC(path, DFS_MAX_PATH);
+	if (path == NULL)
+		return;
+	snprintf(fd_path, sizeof(fd_path) - 1, "/proc/self/fd/%d", fd);
+	len = readlink(fd_path, path, DFS_MAX_PATH - 1);
+	if (len < 0) {
+		DS_ERROR(errno, "readlink() failed");
+		goto out;
+	}
+	path[len] = 0;
+	if (strncmp(path, "socket:", 7) == 0 || strncmp(path, "anon_inode:[eventpoll]", 22) == 0) {
+		D_WARN("dup2/fcntl is closing fd %d (%s) which may be used by DAOS!", fd, path);
```

Comment:
i guess i don't understand what a warning is supposed to do. how can we end up in this situation?

Reply:
This message is mainly for debugging. Not sure whether this is always related to real issues, especially for an application uses network. Shall I use D_DEBUG() instead?
If a bash script uses larger fd (e.g., 15) directly, this piece of code will be triggered.

Reply:
For what it's worth the ioil code is at https://github.com/daos-stack/daos/blob/master/src/client/dfuse/il/int_posix.c#L741-L797

It just uses a hard-coded value of 10 and does no checking.  It also uses dup for all but the first fd and does it all in one function rather than using globals.